scapy/sendrecv.py

```diff
@@ -197,28 +227,34 @@ def sndrcv(pks, pkt, timeout=None, inter=0, verbose=None, chainCC=False,
             timeout = None
         stopevent = threading.Event()
 
+        hsent = {}
+
         thread = threading.Thread(
             target=_sndrcv_snd,
-            args=(pks, timeout, inter, verbose, tobesent, stopevent),
+            args=(pks, timeout, inter, verbose, tobesent, hsent, stopevent),
         )
         thread.start()
 
         hsent, newans, nbrecv, notans = _sndrcv_rcv(
-            (rcv_pks or pks), tobesent, stopevent, nbrecv, notans, verbose, chainCC, multi,
+            (rcv_pks or pks), hsent, stopevent, nbrecv, notans, verbose, chainCC, multi,
+            _storage_policy=_storage_policy,
         )
         thread.join()
         ans.extend(newans)
 
-        remain = list(itertools.chain(*six.itervalues(hsent)))
-        if multi:
-            remain = [p for p in remain if not hasattr(p, '_answered')]
+        if sto_uans:
+            remain = list(itertools.chain(*six.itervalues(hsent)))
```

Comment:
What I don’t like in sndrcv is the usage of lists. What’s good with sto_uans is that it allows to use a ‘no list’ mode, as we don’t care about not answered packets.

However, you are right, I have forgotten to call `process` on unanswered packets , even if I said so in the __doc__

Reply:
(Just saw your edit, OK with it). Will do it later today